src/contracts/core/StakeRootCompendium.sol

```diff
@@ -26,170 +25,139 @@ contract StakeRootCompendium is StakeRootCompendiumStorage {
-        );
+        (bool success, ) = payable(msg.sender).call{value: amount}("");
+        require(success, "StakeRootCompendium.withdrawForOperatorSet: eth transfer failed");
 
-        _updateTotals(numStrategiesBefore, operatorSetToStrategyAndMultipliers[msg.sender][operatorSetId].length());
+        return amount;
     }
 
     /// @inheritdoc IStakeRootCompendium
     function removeStrategiesAndMultipliers(uint32 operatorSetId, IStrategy[] calldata strategies) external {
-        IAVSDirectory.OperatorSet memory operatorSet =
-            IAVSDirectory.OperatorSet({avs: msg.sender, operatorSetId: operatorSetId});
         // update the deposit balance for the operator set whenever number of strategies is changed
-        _updateDepositBalanceInfo(operatorSet, true);
+        IAVSDirectory.OperatorSet memory operatorSet = IAVSDirectory.OperatorSet({avs: msg.sender, operatorSetId: operatorSetId});
+        _updateDepositInfo(operatorSet);
+        uint256 numStrategiesBefore = operatorSetToStrategyAndMultipliers[msg.sender][operatorSet.operatorSetId].length();
+
+        for (uint256 i = 0; i < strategies.length; i++) {
+            require(
+                operatorSetToStrategyAndMultipliers[msg.sender][operatorSet.operatorSetId].remove(address(strategies[i])),
+                "StakeRootCompendium.removeStrategiesAndMultipliers: strategy not found"
+            );
+        }
+
+        _updateTotalStrategies(numStrategiesBefore, operatorSet);
+    }
+
 
-        // remove the strategies and multipliers for the operator set
-        _removeStrategiesAndMultipliers(operatorSet, strategies);
+    function removeOperatorSetFromStakeTree(uint32 operatorSetId) external {
+        _removeOperatorSetFromStakeTree(IAVSDirectory.OperatorSet({avs: msg.sender, operatorSetId: operatorSetId}));
     }
 
     /// @inheritdoc IStakeRootCompendium
     function setOperatorSetExtraData(uint32 operatorSetId, bytes32 extraData) external {
-        (bool exists,, uint224 index) = operatorSetToIndex[msg.sender][operatorSetId].latestCheckpoint();
-        require(exists && index != REMOVED_INDEX, "StakeRootCompendium.setOperatorSetExtraData: operatorSet is not in stakeTree");
         operatorSetExtraDatas[msg.sender][operatorSetId] = extraData;
     }
 
     /// @inheritdoc IStakeRootCompendium
     function setOperatorExtraData(uint32 operatorSetId, address operator, bytes32 extraData) external {
-        (bool exists,, uint224 index) = operatorSetToIndex[msg.sender][operatorSetId].latestCheckpoint();
-        require(exists && index != REMOVED_INDEX, "StakeRootCompendium.setOperatorExtraData: operatorSet is not in stakeTree");
         operatorExtraDatas[msg.sender][operatorSetId][operator] = extraData;
     }
 
     /// POSTING ROOTS AND BLACKLISTING
 
     /// @inheritdoc IStakeRootCompendium
     function verifyStakeRoot(
-        uint32 calculationTimestamp,
+        uint64 calculationBlockNumber,
```

Comment:
this should be a timestamp since everything is in timestamps

Reply:
hmmmm this might be awkward since the zk proof is against a blockhash and blocks are sequential but calculationTimestamps can jump
BN: [ 0 ] [ 1 ] [ 2 ]
TS:   12    36   48

so there is a missing timestamp corresponding to timestamp 24. . . should a proof be skipped in this case?